pkg/operator/encryption/controllers.go

```diff
@@ -62,81 +59,77 @@ func NewControllers(
-				operatorClient,
-				apiServerClient,
-				apiServerInformer,
-				kubeInformersForNamespaces,
-				secretsClient,
-				encryptionSecretSelector,
-				eventRecorder,
-			),
-			controllers.NewStateController(
-				component,
-				provider,
-				deployer,
-				encryptionEnabledChecker.PreconditionFulfilled,
-				operatorClient,
-				apiServerInformer,
-				kubeInformersForNamespaces,
-				secretsClient,
-				encryptionSecretSelector,
-				eventRecorder,
-			),
-			controllers.NewPruneController(
-				component,
-				provider,
-				deployer,
-				encryptionEnabledChecker.PreconditionFulfilled,
-				operatorClient,
-				apiServerInformer,
-				kubeInformersForNamespaces,
-				secretsClient,
-				encryptionSecretSelector,
-				eventRecorder,
-			),
-			controllers.NewMigrationController(
-				component,
-				provider,
-				deployer,
-				encryptionEnabledChecker.PreconditionFulfilled,
-				migrator,
-				operatorClient,
-				apiServerInformer,
-				kubeInformersForNamespaces,
-				secretsClient,
-				encryptionSecretSelector,
-				eventRecorder,
-			),
-			controllers.NewConditionController(
-				component,
-				provider,
-				deployer,
-				encryptionEnabledChecker.PreconditionFulfilled,
-				operatorClient,
-				apiServerInformer,
-				kubeInformersForNamespaces,
-				secretsClient,
-				encryptionSecretSelector,
-				eventRecorder,
-			),
-		},
+	return []factory.Controller{
+		controllers.NewKeyController(
+			component,
+			unsupportedConfigPrefix,
+			provider,
+			deployer,
+			encryptionEnabledChecker.PreconditionFulfilled,
+			operatorClient,
+			apiServerClient,
+			apiServerInformer,
+			kubeInformersForNamespaces,
+			secretsClient,
+			encryptionSecretSelector,
+			eventRecorder,
+		),
+		controllers.NewStateController(
+			component,
+			provider,
+			deployer,
+			encryptionEnabledChecker.PreconditionFulfilled,
+			operatorClient,
+			apiServerInformer,
+			kubeInformersForNamespaces,
+			secretsClient,
+			encryptionSecretSelector,
+			eventRecorder,
+		),
+		controllers.NewPruneController(
+			component,
+			provider,
+			deployer,
+			encryptionEnabledChecker.PreconditionFulfilled,
+			operatorClient,
+			apiServerInformer,
+			kubeInformersForNamespaces,
+			secretsClient,
+			encryptionSecretSelector,
+			eventRecorder,
+		),
+		controllers.NewMigrationController(
+			component,
+			provider,
+			deployer,
+			encryptionEnabledChecker.PreconditionFulfilled,
+			migrator,
+			operatorClient,
+			apiServerInformer,
+			kubeInformersForNamespaces,
+			secretsClient,
+			encryptionSecretSelector,
+			eventRecorder,
+		),
+		controllers.NewConditionController(
+			component,
+			provider,
+			deployer,
+			encryptionEnabledChecker.PreconditionFulfilled,
+			operatorClient,
+			apiServerInformer,
+			kubeInformersForNamespaces,
+			secretsClient,
+			encryptionSecretSelector,
+			eventRecorder,
+		),
 	}, nil
 }
 
-type Controllers struct {
-	controllers []runner
-}
+type Controllers []factory.Controller
```

Comment:
I don't personally care for this construct.  I prefer the struct style this replaced.